### PR TITLE
[Haxe][CodeGenerator] Improvements for code generator, fixed runtime error. fixes #1743

### LINKED ISSUE #1743: [Haxe][ASGenerator] Runtime error `Object reference not set to an instance of an object.` after execution `Generate interface definition` for get|set
```haxe
package;
interface IFoo {
	var foo(g|et, null):haxe.Timer->haxe.ds.Vector<Int>; // | <-- cursor
}
```
```
Object reference not set to an instance of an object.

   at ASCompletion.Completion.ASGenerator.GenerateFunctionJob(GeneratorJobType job, ScintillaControl sci, MemberModel member, Boolean detach, ClassModel inClass) in C:\projects\opensource\flashdevelop\External\Plugins\ASCompletion\Completion\ASGenerator.cs:line 2830
   at ASCompletion.Completion.ASGenerator.GenerateJob(GeneratorJobType job, MemberModel member, ClassModel inClass, String itemLabel, Object data) in C:\projects\opensource\flashdevelop\External\Plugins\ASCompletion\Completion\ASGenerator.cs:line 953
   at ASCompletion.Completion.GeneratorItem.get_Value() in C:\projects\opensource\flashdevelop\External\Plugins\ASCompletion\Completion\ASGenerator.cs:line 4791
   at PluginCore.Controls.CompletionList.ReplaceText(ScintillaControl sci, String tail, Char trigger) in C:\projects\opensource\flashdevelop\PluginCore\PluginCore\Controls\CompletionList.cs:line 810
   at PluginCore.Controls.CompletionList.ReplaceText(ScintillaControl sci, Char trigger) in C:\projects\opensource\flashdevelop\PluginCore\PluginCore\Controls\CompletionList.cs:line 788
   at PluginCore.Controls.CompletionList.HandleKeys(ScintillaControl sci, Keys key) in C:\projects\opensource\flashdevelop\PluginCore\PluginCore\Controls\CompletionList.cs:line 900
   at PluginCore.Controls.UITools.HandleKeys(Keys key) in C:\projects\opensource\flashdevelop\PluginCore\PluginCore\Controls\UITools.cs:line 458
   at PluginCore.Controls.UITools.HandleEvent(Object sender, NotifyEvent e, HandlingPriority priority) in C:\projects\opensource\flashdevelop\PluginCore\PluginCore\Controls\UITools.cs:line 145
   at PluginCore.Managers.EventManager.DispatchEvent(Object sender, NotifyEvent e) in C:\projects\opensource\flashdevelop\PluginCore\PluginCore\Managers\EventManager.cs:line 122
```

### PR DESCRIPTION
**Pull request by SlavaRa**
#1743

### CODE REVIEW

File: External/Plugins/ASCompletion/Completion/ASGenerator.cs [hunk 1]
```diff
@@ -103,24 +102,25 @@ public static void ContextualGenerator(ScintillaControl Sci, List<ICompletionLis
...
+                if ((resolve.Type == null || resolve.Type.IsVoid() || !context.IsImported(resolve.Type, line)) && CheckAutoImport(resolve, options)) return;
                 if (resolve.Type == null)
-                {
+                {
                     suggestItemDeclaration = ASComplete.IsTextStyle(Sci.BaseStyleAt(position - 1));
                 }
-            }
-
+            }
+            var text = Sci.GetLine(line);
```
Comment (Neverbirth) [inline External/Plugins/ASCompletion/Completion/ASGenerator.cs]: Is this needed at all? I don't see it :S

**Comment by Neverbirth** [comment]
Will check this one soon.

**Comment by Neverbirth** [comment]
I fail to see the benefit of those `var text = Sci.GetLine(line);` declarations. I don't think it makes the code more readable or clear and although the existing code makes getting the current line for cases where it's not needed, this change makes getting the line more than once on some exceptional cases (the second and third `text` cases don't always return and they can happen at the same time from what I tested).

So if this is some attempt to optimize I'm not sure it's good, plus there are also several other `Sci.GetLine(` and some `Sci.CurrentLine` and all those ones would be the same as `text` and `line` from what I see?

Also, would be nice to cover this with some tests.

Anyway, if @Meychi decides so this can be merged.

**Comment by SlavaRa** [comment]
@Neverbirth I think that most of the code of this method will be refactoed in the near future

**Comment by Neverbirth** [comment]
If the code is going to be refactored then tests would be even more helpful to catch possible errors during the work, but this would require tons of tests that should be already there and not just the ones here.

**Comment by SlavaRa** [comment]
I'm working on tests at the moment.

**Pull request merged**